util/command_line.py

```diff
@@ -9,6 +11,204 @@
 interactive_console = debug_console
 
 
+from libtbx.phil import parse
+from dxtbx.model.experiment_list import ExperimentListFactory
+from dials.array_family.flex import reflection_table
+
+
+def standard_scope():
+    """Define the standard scope as a function so every instance is unique,
+    not global."""
+    return parse(
+        """
+    input {
+      experiments = None
+        .type = path
+        .multiple = True
+      reflections = None
+        .type = path
+        .multiple = True
+      }
+
+      output {
+      experiments = None
+        .type = path
+        .multiple = True
+      reflections = None
+        .type = path
+        .multiple = True
+      }
+    """
+    )
+
+
+class OptionParser:
+    """Thin wrapper around argparse and phil to get DIALS arguments
+    parsed without actually loading all the data files."""
+
+    @staticmethod
+    def guess_input_file_type(filenames):
+        """Guess if input filenames are experiment-like, reflection-like or
+        param-like, or optionally image like. Return dictionary."""
+
+        experiment_file_endings = ".expt".split()
+        reflection_file_endings = ".refl".split()
+        parameter_file_endings = ".params .phil".split()
+        image_file_endings = ".h5 .nxs .cbf .img".split()
+        image_compression = ["", ".gz", ".bz2"]
+
+        experiment_files = [
+            filename
+            for filename in filenames
+            if any(filename.endswith(exten) for exten in experiment_file_endings)
+        ]
+        reflection_files = [
+            filename
+            for filename in filenames
+            if any(filename.endswith(exten) for exten in reflection_file_endings)
+        ]
+        parameter_files = [
+            filename
+            for filename in filenames
+            if any(filename.endswith(exten) for exten in parameter_file_endings)
+        ]
+        image_files = [
+            filename
+            for filename in filenames
+            if any(
+                filename.endswith(exten + compression)
+                for exten in image_file_endings
+                for compression in image_compression
+            )
+        ]
+
+        unknown = [
+            filename
+            for filename in filenames
+            if not any(
+                (
+                    filename in experiment_files,
+                    filename in reflection_files,
+                    filename in parameter_files,
+                    filename in image_files,
+                )
+            )
+        ]
+
+        return {
+            "experiments": experiment_files,
+            "reflections": reflection_files,
+            "parameters": parameter_files,
+            "images": image_files,
+            "unknown": unknown,
+        }
+
+    def __init__(self, phil=None):
+        self._input_experiments = None
+        self._input_reflections = None
+
+        parser = argparse.ArgumentParser()
```

Comment:
Is there a reason why we don't inherit `argparse.ArgumentParser` as a superclass of this `OptionParser`?